### DISCUSSION TITLE
Setting a selection to an inline node

### DISCUSSION THREAD

**Discussion by jackbuehner** [ueberdosis/tiptap#1377] [General Questions] [answered]
I am creating an extension that adds a comment node. I have added `group: 'inline'`, `inline: true`, and `content: 'text*'` to make it an inline node. However, I am unable to set the selection to the comment node with `commands.setNode('comment')`. Additionally, `editor.can().setNode('comment')` always returns `false`. Am I supposed to be using a different command?

I am trying to use an inline node instead of a mark because I need to use [node views](https://www.tiptap.dev/guide/node-views) to store the comment message while highlighting the text connected to the comment message. Note that I have not yet added a node view.

Here is a sandbox with the extension: https://codesandbox.io/s/tiptap-comment-inline-node-issue-uwu6c

**Comment by Rodeoclash** [comment]
Also running into this, did you figure it out??

**Comment by sereneinserenade** [comment, marked as answer]
@jackbuehner You can take a look at the comment extension I created https://github.com/sereneinserenade/tiptap-comment-extension, demo: https://sereneinserenade.github.io/tiptap-comment-extension/, it uses marks and you don't need to have inline nodeview.

and to answer your question, you can add a prosemirror plugin, which will select the whole mark on click. take a look at this https://github.com/sereneinserenade/tiptap-comment-extension/blob/main/src/components/extension/comment.ts#L76-L89

**Reply by jackbuehner** [reply]
I actually ended up doing something very similar. I should have posted an update. Thank you for sharing this solution for whoever finds this in the future.
